syntax = "proto3";

message SumRequest {
  double a = 1;
  double b = 2;
}

message SumReply {
  double s = 1;
}

message MultiplyRequest {
  double a = 1;
  double b = 2;
}

message MultiplyReply {
  double s = 1;
}

message MaxRequest {
  double a = 1;
  double b = 2;
  double c = 3;
}

message MaxReply {
  double s = 1;
}

message DivRequest {
  double a = 1;
  double b = 2;
}

message DivReply {
  double q = 1;
  double r = 2;
}

service Calculator {
  rpc Sum (SumRequest) returns (SumReply);
  rpc Multiply (MultiplyRequest) returns (MultiplyReply);
  rpc Max (MaxRequest) returns (MaxReply);
  rpc Division (DivRequest) returns (DivReply);
}
